video/out/opengl/hwdec_cuda.c

```diff
@@ -103,9 +115,123 @@ static int check_cu(struct ra_hwdec *hw, CUresult err, const char *func)
 
 #define CHECK_CU(x) check_cu(hw, (x), #x)
 
+static bool cuda_ext_gl_init(struct ra_hwdec_mapper *mapper,
+                             const struct ra_format *format, int n);
+static bool cuda_ext_vk_init(struct ra_hwdec_mapper *mapper,
+                             const struct ra_format *format, int n);
+static void cuda_ext_gl_uninit(struct ra_hwdec_mapper *mapper, int n);
+static void cuda_ext_vk_uninit(struct ra_hwdec_mapper *mapper, int n);
```

Comment:
At this point you can probably move the implementation of these functions to separate (conditionally compiled) files, and make a single unifying API, like

```c
struct cuda_ext {
    bool (*init)(...);
    void (*uninit)(....);
}
```

And replace the `is_gl` and `is_vk` bools by a `const struct cuda_ext *` pointer. That way we get rid of basically all ifdef in the code and make the platform-specific code a bit clearer.

Reply:
I actually started off trying to do this, but it became a mess with the handling of the 'priv' structs, and trying to handle all of the vulkan vs gl logic. I can try and do it again, but I'll handle it as a separate PR.